aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -138,241 +365,225 @@ def export(
+
+    extract_time = output_data['time_extract_start'] - output_data['time_extract_stop']
+    EXPORT_LOGGER.debug(f'Data extracted in {extract_time:6.2g} s.')
+
+    if export_data is not None:
+        try:
+            output_data['time_write_start'] = time.time()
+            output_data['writer'] = writer.write(export_data=export_data, silent=silent)  # type: ignore
+            output_data['time_write_stop'] = time.time()
+        except (exceptions.ArchiveExportError, LicensingException) as exc:
+            if os.path.exists(filename):
+                os.remove(filename)
+            raise exc
+
+        write_time = output_data['time_write_start'] - output_data['time_write_stop']
+        EXPORT_LOGGER.debug(f'Data written in {write_time:6.2g} s.')
 
-    if file_format == ExportFileFormat.TAR_GZIPPED:
-        file_format_verbose = 'Gzipped tarball (compressed)'
-    # Must be a zip then
-    elif use_compression:
-        file_format_verbose = 'Zip (compressed)'
-    else:
-        file_format_verbose = 'Zip (uncompressed)'
-    summary(file_format_verbose, filename, **kwargs)
-
-    try:
-        if file_format == ExportFileFormat.TAR_GZIPPED:
-            times = export_tar(entities=entities, filename=filename, silent=silent, **kwargs)
-        else:  # zip
-            times = export_zip(
-                entities=entities, filename=filename, use_compression=use_compression, silent=silent, **kwargs
-            )
-    except (exceptions.ArchiveExportError, LicensingException) as exc:
-        if os.path.exists(filename):
-            os.remove(filename)
-        raise exc
-
-    if len(times) == 2:
-        export_start, export_end = times  # pylint: disable=unbalanced-tuple-unpacking
-        EXPORT_LOGGER.debug('Exported in %6.2g s.', export_end - export_start)
-    elif len(times) == 4:
-        export_start, export_end, compress_start, compress_end = times
-        EXPORT_LOGGER.debug(
-            'Exported in %6.2g s, compressed in %6.2g s, total: %6.2g s.', export_end - export_start,
-            compress_end - compress_start, compress_end - export_start
-        )
     else:
-        EXPORT_LOGGER.debug('No information about the timing of the export.')
+        EXPORT_LOGGER.debug('No data to write.')
 
     # Reset logging level
     if silent:
-        logging.disable(level=logging.NOTSET)
+        logging.disable(logging.NOTSET)
 
+    return output_data
 
-def export_zip(entities=None, filename=None, use_compression=True, **kwargs):
-    """Export in a zipped folder
+
+@override_log_formatter('%(message)s')
+def _generate_data(
+    entities: Optional[Iterable[Any]] = None,
+    allowed_licenses: Optional[Union[list, Callable]] = None,
+    forbidden_licenses: Optional[Union[list, Callable]] = None,
+    silent: bool = False,
+    include_comments: bool = True,
+    include_logs: bool = True,
+    **traversal_rules: bool,
+) -> Optional[ExportData]:
+    """Generate data to be exported
```

Comment:
```suggestion
    """Collect data to be exported
```

Reply:
fixed in b689f5e6986c15ab18ad225e5659f1ef3bf8f9c0